### ISSUE TITLE
Make the eraser tool's size changeable?

### ISSUE THREAD

**Issue by zacx-z** [mapeditor/tiled#391]
I get frustrated if I would erase all the sky tiles in one layer and create a new 'sky' layer.
I searched the document and didn't find a way to change the size of the eraser tool.

Oh... I've found a method just now. I could select and cut it using COMMAND+X - It's weird that I can't select and just delete it.

I use tiled 0.9.0 on Mac.

**Comment by bjorn** [comment]
Several enhancements have been made since this issue was opened:

* Delete key can be used to delete selected area (no need to use Cut)
* A magic wand selection tool was added
* The eraser now has a right-click-drag mode for quickly erasing rectangular areas

I wonder if I should close this issue, or whether it still makes sense to actually allow the size of the eraser to be changed. Implementing that mainly raises the question of where to put it in the UI, which probably needs some generic place for tool-specific settings (which would be useful for some other tools as well).

**Comment by AxiomVerge** [comment]
I just ran into a situation where I am editing such large chunks of the map at once that a resizable eraser would make it easier.  Basically I am zoomed out far enough that it is almost like editing a bitmap in photoshop.  I notice the "Execute Command" button has a dropdown next to it; maybe the eraser could have one with options like "1x, 2x, 4x, 8x".  Otherwise you could end up going down the rabbit hole of each tool having its own options toolbar like Photoshop (which could be useful but is probably more work than it's worth)

**Comment by bjorn** [comment]
@AxiomVerge Each tool already has its own options tool bar, so adding this option just for the eraser for now would be possible. Does in your case the right-click rectangular area erase mode not help?

**Comment by AxiomVerge** [comment]
> @AxiomVerge Each tool already has its own options tool bar, so adding this option just for the eraser for now would be possible. Does in your case the right-click rectangular area erase mode not help?

No, I'm basically trying to sculpt large areas of tiles by swiping my mouse in a curving motion, so the rectangular shape isn't that helpful.

**Comment by bjorn** [comment]
@AxiomVerge If you use a recent development snapshot, you can add an eraser of any size by adding a custom script like the following:
```js
/*
 * Example big (5x5) eraser tool.
 */
let eraser = tiled.registerTool('BigEraser', {
    name: "Big Eraser",

    doErase(x, y) {
        for (let _x = -2; _x < 3; ++_x)
            for (let _y = -2; _y < 3; ++_y)
                this.edit.setTile(x + _x, y + _y, null)
        this.edit.apply()
    },

    activated() { this.edit = null },
    tilePositionChanged(x, y) {
        if (this.edit)
            this.doErase(x, y)
    },
    mousePressed(button, x, y, modifiers) {
        const tileLayer = this.map.currentLayer
        if (!tileLayer.isTileLayer)
            return

        this.edit = tileLayer.edit()
        this.doErase(this.tilePosition.x, this.tilePosition.y)
    },
    mouseReleased(button, x, y, modifiers) {
        this.edit = null
    },
})
```
Unfortunately the tool will have no preview to show which area will get erased. I realized the scripting API currently does not have any way to set up a preview for removing tiles, only for setting tiles.

To use this script:

* Open _Preferences > Plugins_ and click the _Open..._ button to open the extensions folder in your file manager.
* Create a sub-folder there for this extension.
* Add a file in that sub-folder with the above script as contents (say `bigeraser.js`).
* Restart Tiled (in the future I'll make sure adding files triggers an automatic reload).

You should then see a "Big Eraser" entry in the tool bar, which allows you to erase 5x5 sections. Just note that since it's a bit simplistic, it will skip areas if you move your mouse very fast (whereas the built-in eraser tool will draw a line from the previous location).

I hope this helps for now.

**Comment by AxiomVerge** [comment]
Wow, that is VERY cool.  I can't wait to see what other kind of scripts show up once this feature is in the release build.  THANKS!!
